README.md

```diff
@@ -141,6 +150,46 @@ end
     end
 ```
 
+## Properties Applying to All InSpec AWS Resources
+
+### `aws_region`
+
+In order to provide multi-region support, the `aws_region` property may be specified to a resource.  This will only have an effect on AWS resources that have a region dependency e.g. security groups.  One special-case worth mentioning is the `aws_s3_bucket` resource that updates its region based on the location returned from S3.
+
+The `aws_regions` resource can be used to loop across all regions e.g.
+
+```
+  aws_regions.region_names.each do |region|
+    <use region in other resources here>
+  end
+```
+
+### `aws_endpoint`
+
+A custom endpoint URL can optionally be specified to resources for testing other compatible providers.  This propagates to the AWS client configuration.  An example is provided below for [Minio](https://github.com/minio/minio) S3 compatible buckets e.g.
+
+```
+title 'Test For  Minio Buckets Existing at a Custom Endpoint'
+
+endpoint = attribute(:minio_server, default: 'http://127.0.0.1:9000', description: 'The Minio server custom endpoint.')
+
+control 'minio-buckets-1.0' do
+  impact 1.0
+  title 'Ensure Minio buckets exist.'
+
+  describe aws_s3_bucket(aws_endpoint: endpoint, bucket_name: 'miniobucket') do
+    it { should exist }
+  end
+
+  describe aws_s3_bucket(aws_endpoint: endpoint, bucket_name: 'notthere') do
+    it { should_not exist }
+  end
+end
+```
+
+Note that InSpec AWS assumes full compatibility with the underlying AWS SDK and unsupported operations will cause failures.  Therefore, depending on the external provider implementation your mileage may vary!
```

Comment:
+1 for this. May want to add something along the lines of 

>For example, Minio currently [does not support all of the S3 APIs](https://github.com/minio/minio/blob/master/docs/minio-limits.md#list-of-amazon-s3-bucket-apis-not-supported-on-minio), although InSpec AWS itself may.

Reply:
Thanks, the supported APIs (also what we depend on this side) could change so I'll probably err on the side of leaving out the low-level detail.